include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -1185,7 +1185,7 @@ __pattern_merge(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _Ite
 
         __par_backend_hetero::__parallel_merge(_BackendTag{}, ::std::forward<_ExecutionPolicy>(__exec),
                                                __buf1.all_view(), __buf2.all_view(), __buf3.all_view(), __comp)
-            .wait();
+            .__deferrable_wait();
```

Comment:
`__pattern_inplace_merge` requires this pattern to by synchronous, so the merge is finished before the copy.

Reply:
@SergeyKopienko please take into account.

Reply:
@akukanov, @danhoeflinger Fixed.